rsrcdump/resfork.py

```diff
@@ -43,17 +44,17 @@ def desc(self) -> str:
         return f"{sanitize_type_name(self.type)}#{self.num}"
 
     @property
-    def type_str(self, errors='replace') -> str:
+    def type_str(self, errors: str='replace') -> str:
         return self.type.decode(get_global_encoding(), errors)
 
     @property
-    def name_str(self, errors='replace') -> str:
+    def name_str(self, errors: str='replace') -> str:
         return self.name.decode(get_global_encoding(), errors)
 
 
 @dataclass
 class ResourceFork:
-    tree: dict[ResType, dict[int, Resource]] = field(default_factory=dict)
+    tree: dict[ResType, dict[str | bytes | int, Resource]] = field(default_factory=dict)
```

Comment:
The inner dict should be `dict[int, Resource]`. Conceptually, it maps resource IDs (always an int) to a Resource object.

Reply:
Fixed in e907f63